### PR TITLE
Add function definitions

### PR DESCRIPTION
**Pull request by non-binary**
Ready for Review. See most recent comment.

### CODE REVIEW

**Comment by non-binary** [comment]
@zkat I feel silly asking this but I'm having a hard time figuring out how to port these objects/dictionaries to Rust.

I tried using a HashMap but they don't like to be initialized in mod scope I think, or at least I'm doing it incorrectly lol. How would you define these in Rust? If you could help me figure out one of them, I can write out the second one.

```
const hopByHopHeaders = {
    date: true, 
    connection: true,
    'keep-alive': true,
    'proxy-authenticate': true,
    'proxy-authorization': true,
    te: true,
    trailer: true,
    'transfer-encoding': true,
    upgrade: true,
};

const excludedFromRevalidationUpdate = {
    'content-length': true,
    'content-encoding': true,
    'transfer-encoding': true,
    'content-range': true,
};
```

Thanks!

**Comment by zkat** [comment]
I'm not sure what the best way is, but I suspect it probably has to do with using [`lazy_static`](https://crates.io/crates/lazy_static) to define these as [`HashSet`](https://doc.rust-lang.org/std/collections/struct.HashSet.html)s.

I haven't used this before but I think those structs would look like:

```
use std::collections::HashSet;
lazy_static! {
    static ref HOP_BY_HOP_HEADERS: HashSet<&'static str> = {
        let mut set = HashSet::new();
        set.insert("date");
        ....
        set
    };
}
```

Wanna give that a whirl?

File: src/lib.rs [hunk 1]
```diff
@@ -1,7 +1,136 @@
...
+
+    // excluded_from_revalidation_update
+
+    fn parse_cache_control() -> () {
+        return ()
     }
-}
+
+    fn format_cache_control() -> () {
+        return ()
```
Comment (zkat) [inline src/lib.rs]: I like using `unimplemented!();` in places like these instead of empty returns.

File: src/lib.rs [hunk 2]
```diff
@@ -1,7 +1,136 @@
...
+        302,
+        303,
+        307,
+        308,
+        404,
+        405,
+        410,
+        414,
+        501,
+    ];
```
Comment (zkat) [inline src/lib.rs]: These might be worth turning into lazy_static HashSets too, idk. Better than a linear search?

File: src/lib.rs (hunk 1 above)
Comment (non-binary) [reply src/lib.rs]: Ah okay, didn't know this exists. Thanks! I'll change them over.

File: src/lib.rs (hunk 2 above)
Comment (non-binary) [reply src/lib.rs]: Sure!

**Comment by non-binary** [comment]
@zkat Was getting started with `HashSet` and wouldn't that mean it would just be a set of strings `[date, connection, keep-alive, proxy-authenticate, proxy-authorization, te, trailer, transfer-encoding, upgrade]` without the `true` values? Are they not necessary?

**Comment by non-binary** [comment]
@chrisdickinson Any thoughts on what's above? Thanks!

**Comment by zkat** [comment]
@non-binary I'm pretty sure those true values are just being used because the object is being used as a set, which is why I recommended the change I did. The use case is just looking up if those keys exist.

**Comment by non-binary** [comment]
Ah, that makes sense. Just wanted to clarify. Thanks!

**Comment by non-binary** [comment]
Just need to add the empty/failing tests and then this PR will be ready to review. :party_parrot:

**Comment by zkat** [comment]
I'm on a flight now so I'm not gonna be able to review things until later tonight, btw

**Comment by non-binary** [comment]
No problem/pressure! I may not even get to the rest of it today. Take your time. :)

**Comment by non-binary** [comment]
@zkat Ready for review! I've added definitions for all of the functions and tests present in the JS version. Please let me know if you think any of this could be done better. Anything helps!

**Comment by non-binary** [comment]
Yay!

**Pull request merged**